backup/restore/restore.go

```diff
@@ -0,0 +1,138 @@
+package restore
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/codeskyblue/go-sh"
+	"github.com/pkg/errors"
+	"github.com/stefanprodan/mgob/config"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	host = "127.0.0.1"
+	port = 27017
+)
+
+// Responsible to restore backup from one source
+// using mongorestore
+// Responsibilities
+// - Download backup from one source
+// - Restore backup using mongorestore
+// - Testing restoring using queries defined by plan
+func Restore(plan config.Plan, archive string) (string, error) {
+	err := startMongoToRestore()
+	if err != nil {
+		return "", err
+	}
+	defer shutdownMongo()
+	restore := fmt.Sprintf("mongorestore --archive=%v --gzip --host %v --port %v ",
+		archive, host, port)
+	if plan.Target.Database != "" {
+		restore += fmt.Sprintf("--db %v ", plan.Target.Database)
+	}
+
+	output, err := sh.Command("/bin/sh", "-c", restore).SetTimeout(time.Duration(plan.Scheduler.Timeout) * time.Minute).CombinedOutput()
+	if err != nil {
+		ex := ""
+		if len(output) > 0 {
+			ex = strings.Replace(string(output), "\n", " ", -1)
+		}
+		return "", errors.Wrapf(err, "mongorestore log %v", ex)
+	}
+	fmt.Printf("%s\n", output)
+	checkMsg, err := checkRestore(host, port, plan.Restore.Database, plan.Restore)
+	if err != nil {
+		return "", err
+	}
+	return checkMsg, nil
+}
+
+func cleanMongo(s *mgo.Session) error {
+	names, err := s.DatabaseNames()
+	if err != nil {
+		return err
+	}
+	for _, name := range names {
+		err = s.DB(name).DropDatabase()
+		fmt.Printf("droping database  = %v\n", name)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkRestore(host string, port int, database string, r config.Restore) (string, error) {
+	mongoURL := fmt.Sprintf("mongodb://%v:%d", host, port)
+	session, err := mgo.Dial(mongoURL)
+	if err != nil {
+		return "", err
+	}
+	defer session.Close()
+	defer cleanMongo(session)
+
+	session.SetSafe(&mgo.Safe{})
+	for _, collec := range r.Collections {
+		c := session.DB(database).C(collec.Name)
+		countRestored, err2 := c.Find(bson.M{}).Count()
```

Comment:
maybe use `var countRestored int`  for reuse the var err and don't need err2

Reply:
yeap :)